src/System.Drawing.Common/src/mono/System.Drawing.Drawing2D/GraphicsPath.cs

```diff
@@ -0,0 +1,896 @@
+//
+
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+
+namespace System.Drawing.Drawing2D
+{
+	public sealed class GraphicsPath : MarshalByRefObject, ICloneable, IDisposable
+	{
+		// 1/4 is the FlatnessDefault as defined in GdiPlusEnums.h
+		private const float FlatnessDefault = 1.0f / 4.0f;
+
+		internal IntPtr nativePath = IntPtr.Zero;
+
+		GraphicsPath (IntPtr ptr)
+		{
+			nativePath = ptr;
+		}
+
+		public GraphicsPath ()
+		{
+                        Status status = SafeNativeMethods.Gdip.GdipCreatePath (FillMode.Alternate, out nativePath);
+                        SafeNativeMethods.Gdip.CheckStatus (status);
+		}
+
+		public GraphicsPath (FillMode fillMode)
+		{
+                        Status status = SafeNativeMethods.Gdip.GdipCreatePath (fillMode, out nativePath);
+                        SafeNativeMethods.Gdip.CheckStatus (status);
+		}
+
+		public GraphicsPath (Point[] pts, byte[] types)
+			: this (pts, types, FillMode.Alternate)
+		{
+		}
+
+		public GraphicsPath (PointF[] pts, byte[] types)
+			: this (pts, types, FillMode.Alternate)
+		{
+		}
+
+		public GraphicsPath (Point[] pts, byte[] types, FillMode fillMode)
+		{
+			if (pts == null)
+				throw new ArgumentNullException ("pts");
+			if (pts.Length != types.Length)
+				throw new ArgumentException ("Invalid parameter passed. Number of points and types must be same.");
+
+			Status status = SafeNativeMethods.Gdip.GdipCreatePath2I (pts, types, pts.Length, fillMode, out nativePath);
+			SafeNativeMethods.Gdip.CheckStatus (status);
+		}
+
+		public GraphicsPath (PointF[] pts, byte[] types, FillMode fillMode)
+		{
+			if (pts == null)
+				throw new ArgumentNullException ("pts");
+			if (pts.Length != types.Length)
+				throw new ArgumentException ("Invalid parameter passed. Number of points and types must be same.");
+
+			Status status = SafeNativeMethods.Gdip.GdipCreatePath2 (pts, types, pts.Length, fillMode, out nativePath);
+			SafeNativeMethods.Gdip.CheckStatus (status);
+		}
+	
+                public object Clone ()
+                {
+                        IntPtr clone;
+
+                        Status status = SafeNativeMethods.Gdip.GdipClonePath (nativePath, out clone);
+                        SafeNativeMethods.Gdip.CheckStatus (status);                      	
+
+                        return new GraphicsPath (clone);
+                }
+
+                public void Dispose ()
+                {
+                        Dispose (true);
+                        System.GC.SuppressFinalize (this);
+                }
+
+                ~GraphicsPath ()
+                {
+                        Dispose (false);
+                }
+                
+		void Dispose (bool disposing)
+		{
+			Status status;
+			if (nativePath != IntPtr.Zero) {
+				status = SafeNativeMethods.Gdip.GdipDeletePath (nativePath);
+				SafeNativeMethods.Gdip.CheckStatus (status);
+
+				nativePath = IntPtr.Zero;
+			}
+		}
+
+		public FillMode FillMode {
+			get {
```

Comment:
Nit: brace style doesn't match that used elsewhere in corefx.  Hopefully codeformatter would address that.